go/state/gostate/archive_state.go

```diff
@@ -155,14 +155,21 @@ func (s *ArchiveState) Export(ctx context.Context, out io.Writer) (common.Hash,
 		return common.Hash{}, state.ExportNotSupported
 	}
 
-	exportableTrie := mptio.NewExportableArchiveTrie(trie, s.block)
-	rootHash, err := mptio.ExportLive(ctx, mptio.NewLog(), exportableTrie, out)
+	err := mptio.ExportBlockFromOnlineArchive(ctx, trie, out, s.block)
+	if err != nil {
+		// export opens the trie parallel to the current program,
+		// possible error is sent to the caller but should not corrupt
+		// the database on error.
+		return common.Hash{}, err
+	}
+
+	rootHash, err := trie.GetHash(s.block)
 	if err != nil {
 		s.archiveError = errors.Join(s.archiveError, err)
 		return common.Hash{}, s.archiveError
 	}
 
-	return rootHash, s.archiveError
+	return rootHash, nil
```

Comment:
Is the error really `nil` in this case? I think it should remain to be the `s.archiveError` ...

Reply:
I think it is - if it was not nil, this method would return at the very beginning.

Reply:
oh, right. it must have leaked during merge